### PR TITLE
Fix shadowsocks/shadowsocks-android#2562

### PR DESCRIPTION
**Pull request by madeye**
Fix https://github.com/shadowsocks/shadowsocks-android/issues/2562

### CODE REVIEW

**Comment by Mygod** [comment]
> The latest openssl 0.10.30 pakcage requires a minimal SDK target 23

Why is this?

Also there is a better fix imo for this.

**Comment by zonyitoo** [comment]
> The latest openssl 0.10.30 pakcage requires a minimal SDK target 23

That's bad.. :(

**Comment by Mygod** [comment]
https://github.com/shadowsocks/shadowsocks-android/commit/0d75659ce6f2d4a8e125c5d10c53a508b28fcdad

We should try to find a fix for 0.10.30 as well.

**Comment by madeye** [comment]
Maybe we just increase the minSDK to 23...

**Comment by Mygod** [comment]
Where did you find the information regarding minSDK?

**Comment by RebelliousWhiz** [comment]
I don't know what's impacting the code, but to my perspective, people still use SDK 23 devices don't care about GFW.... My grandma has a MI 6, and it's now on Android 10...

I guess it's completely fine to set minSDK to 23.

**Comment by Mygod** [comment]
Reported to upstream: https://github.com/sfackler/rust-openssl/issues/1315

If we do not get a satisfactory fix we can drop support for Android 5.

**Comment by zonyitoo** [comment]
So, is it ok for me to merge this MR? @Mygod

**Comment by Mygod** [comment]
I don't think we need this merged for now.

File: src/plugin/ss_plugin.rs [hunk 1]
```diff
@@ -21,14 +21,18 @@ pub fn plugin_cmd(plugin: &PluginConfig, remote: &ServerAddr, local: &SocketAddr
...
-                cmd.arg("-V");
-            }
-        }
+		if cfg!(target_os = "android")
+		{
+			let mut tmp = opt.clone();
+			// Add VPN flags to the commandline as well
+			if tmp.contains(";V") {
+				cmd.arg("-V");
+				tmp = tmp.replace(";V", "")
```
Comment (zonyitoo) [inline src/plugin/ss_plugin.rs:30]: BTW, `;V` parameter is only for v2ray-plugin. This strategy should only be applied to this specific plugin, right?

**Comment by madeye** [comment]
I think the missing part is how to pass VPN option to plugins, which is not defined before.

Maybe we should fix this from plugin's side instead.

File: src/plugin/ss_plugin.rs (hunk 1 above)
Comment (madeye) [reply src/plugin/ss_plugin.rs:30]: Not sure how third-party plugins handle these options. But currently we only see v2ray-plugin has issues to parse `;V` options.

**Comment by Mygod** [comment]
So let's settle this once and for all: what should we use? `-V` or `V` in plugin options.

**Comment by zonyitoo** [comment]
Since `SS_PLUGIN_OPTIONS` is defined as "options for plugin", the VPN flag should be passed in options.

**Comment by zonyitoo** [comment]
Or maybe we can make another new environment variable, such as `SS_PLUGIN_CTRL`, which could be used for passing standardized controlling options, like `vpn_mode=1&verbose=3&fast_open=1`

**Comment by madeye** [comment]
Let's put `V` into the `SS_PLUGIN_OPTIONS`. I'll release a new `v2ray-plugin` to ensure the VPU flags is parsed correctly.

**Comment by madeye** [comment]
Fixed via https://github.com/shadowsocks/v2ray-plugin-android/commit/47888430131386edfa7c635e324796ed0357b85b

**Comment by Mygod** [comment]
Should we get rid of this hack then?

https://github.com/shadowsocks/shadowsocks-rust/blob/efd9e7747db5b1f924943cc29e570c877167b476/src/plugin/ss_plugin.rs#L28

**Comment by madeye** [comment]
Yep, let's remove that hack.

**Pull request closed without merging**